File: Buoi1/vidu/vd2/vd2.cpp
#include <iostream>
#include <vector>
using namespace std;

int linear_search(const vector<int>& arr, int target) {
    for (size_t index = 0; index < arr.size(); index++) {
        if (arr[index] == target) {
            return index; // Trả về chỉ số nếu tìm thấy phần tử
        }
    }
    return -1; // Trả về -1 nếu không tìm thấy
}

int main() {
    vector<int> arr = {4, 2, 5, 1, 3};
    cout << linear_search(arr, 5) << endl; // Output: 2
    return 0;
}

File: Buoi1/vidu/vd2/vd2.jl
function linear_search(arr, target)
    for (index, value) in enumerate(arr)
        if value == target
            return index - 1  # Trả về chỉ số (trừ 1 vì Julia đánh chỉ số từ 1)
        end
    end
    return -1  # Trả về -1 nếu không tìm thấy
end

arr = [4, 2, 5, 1, 3]
println(linear_search(arr, 5))  # Output: 2
